owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -1060,53 +1062,115 @@ class FileDisplayActivity : FileActivity(),
+                            } else {
+                                R.string.folder_already_exists_description
+                            }, file.fileName
                         )
                     )
-                    .setNeutralButton(R.string.welcome_feature_skip_button) { _, _ ->
-                        replace.add(null)
-                        pos++
-                        if (pos == data.size) {
-                            launchAction(uiResult.data, replace)
+                    .setCheckboxText(this.getString(R.string.apply_to_all_conflicts, countDownValue.toString()))
+                    .build()
+
+
+                if (countDownValue > 1) {
+                    customDialog.checkboxVisible = true
+                }
+
+                customDialog.show(fragmentManager!!, "customdialog")
+
+                openDialogs.add(customDialog)
+
+                customDialog.setDialogButtonClickListener(object : FileAlreadyExistsDialog.DialogButtonClickListener {
+
+                    override fun onKeepBothButtonClick() {
+                        if (customDialog.isCheckBoxChecked) {
+                            repeat(data.asReversed().size) {
+                                replace.add(false)
+                                pos++
+                                if (pos == data.size) {
+                                    launchAction(uiResult.data, replace)
+                                }
+                            }
+                            dismissAllOpenDialogs()
+                        } else {
+                            replace.add(false)
+                            pos++
+                            if (pos == data.size) {
+                                launchAction(uiResult.data!!, replace)
+                            }
+                            customDialog.dismiss()
                         }
                     }
-                    .setNegativeButton(R.string.conflict_replace) { _, _ ->
-                        replace.add(true)
-                        pos++
-                        if (pos == data.size) {
-                            launchAction(uiResult.data, replace)
+
+                    override fun onSkipButtonClick() {
+                        if (customDialog.isCheckBoxChecked) {
+                            repeat(data.asReversed().size) {
+                                replace.add(null)
+                                pos++
+                                if (pos == data.size) {
+                                    launchAction(uiResult.data!!, replace)
+                                }
+                            }
+                            dismissAllOpenDialogs()
+                        } else {
+                            replace.add(null)
+                            pos++
+                            if (pos == data.size) {
+                                launchAction(uiResult.data!!, replace)
```

Comment:
Why we need the '!!' here, in previous versión we didn't use it.

Reply:
Done